### PR TITLE
Removing requirements / links features from the REPL

### PR DESCRIPTION
**Pull request by lgalabru**
With this PR, we are getting rid of the requirements, we came to the conclusion that a more natural place would be clarinet.
There a few other adjustments / additions / fixes that were made along the way of working on https://github.com/hirosystems/clarinet/pull/333, that are required and expected by this PR.

### CODE REVIEW

File: src/clarity/diagnostic.rs [hunk 1]
```diff
@@ -71,55 +71,55 @@ impl Diagnostic {
...
             let first_line = span.start_line.saturating_sub(1) as usize;
             let last_line = span.end_line.saturating_sub(1) as usize;
 
             output.push(lines[first_line].clone());
-            let mut pointer = format!(
-                "{: <1$}^",
-                "",
-                (span.start_column.saturating_sub(1)) as usize
-            );
+            let mut pointer = format!("{: <1$}^", "", (span.start_column - 1) as usize);
```
Comment (obycode) [inline src/clarity/diagnostic.rs:109]: What is the reason for this change from `saturating_sub(1)` to `- 1`. I think it was using the saturating sub so that it correctly handles the case where the start_column is 0.

File: src/repl/session.rs [hunk 2]
```diff
@@ -536,6 +343,47 @@ impl Session {
...
+        );
+
+        let bns_contract = ContractName::try_from("bns").unwrap();
+        let pox_contract = ContractName::try_from("pox").unwrap();
+
+        let boot_testnet =
+            PrincipalData::parse_standard_principal("ST000000000000000000002AMW42H").unwrap();
+
+        let boot_mainnet =
+            PrincipalData::parse_standard_principal("SP000000000000000000002Q6VF78").unwrap();
```
Comment (obycode) [inline src/repl/session.rs:366]: Would it make sense to move these into lazy statics?

**Review by obycode — APPROVED** [review]
Looks good to me, just two questions.

File: src/clarity/diagnostic.rs (hunk 1 above)
Comment (lgalabru) [reply src/clarity/diagnostic.rs]: Addressed with [latest](https://github.com/hirosystems/clarity-repl/pull/159/commits/257b12af8673c56d8112232e2489b564beaf5755)

File: src/repl/session.rs (hunk 2 above)
Comment (lgalabru) [reply src/repl/session.rs]: Addressed with [latest](https://github.com/hirosystems/clarity-repl/pull/159/commits/257b12af8673c56d8112232e2489b564beaf5755)

**Review by obycode — APPROVED** [review]
LGTM!

**Review by obycode — APPROVED** [review]
